lib/express-slow-down.js

```diff
@@ -59,6 +60,14 @@ function SlowDown(options) {
         delay = Math.min(unboundedDelay, options.maxDelayMs);
       }
 
+      if (
+        options.maxDelayMs &&
+        options.onMaxDelayReached &&
+        delay === options.maxDelayMs
```

Comment:
i think after reaching the time maxDelay we are kind of missing the point of slow-down 
because in case of a lot of requests getting together they all eventually execute together 

about the new property `req.slowDown.maxDelayReached` for my opinion it called be easily calculated by using 
`req.slowDown.delay` and the value of maxDelay pass to options.

what do you think about the other approach  , instead this new call-back function just to clear
the setTimeout in case the request is closed?

instead of 

`
   if (delay !== 0) {
        return setTimeout(next, delay);
      }
`

```
      if (delay !== 0) {
        const timerId = setTimeout(next, delay);
        req.on("close",function(){
          clearTimeout(timerId);
        });
        req.on("finish",function(){
          clearTimeout(timerId);
        });
        return timerId;
      }

```

Reply:
Sorry for the slow response, it's been a busy week.

I was thinking that this library would generally be combined with express-rate-limit, so after a certain point, it just blocks the requests instead of slowing them down. (You can define a custom handler in express-rate-limit, so "block" can be anything you want.)

Alternatively, when not using a rate-limiter in conjunction with express-slow-down, then just don't set a `maxDelay` and you won't get the bunching together problem.

That delay checking seems like a good idea, although I'm curious what happens right now if the connection is closed by the other side during the delay. If that throws an error, then we should definitely add your suggestion.